### ISSUE TITLE
Why use SVD in Ridge Regression?

### ISSUE THREAD

**Issue by 0shimax** [marcotcr/lime#169]
LIME use Ridge Regression with argument "solver='auto'"  in lime_base.py and in most cases SVD is used as solver. SVD is unstable for very sparse data. I think should we use "solver='sag'" or "solver='saga'".
Modified code is below.

```Python
    def feature_selection(self, data, labels, weights, num_features, method):
        """Selects features for the model. see explain_instance_with_data to
           understand the parameters."""
        if method == 'none':
            return np.array(range(data.shape[1]))
        elif method == 'forward_selection':
            return self.forward_selection(data, labels, weights, num_features)
        elif method == 'highest_weights':
            clf = Ridge(alpha=0, fit_intercept=True,
                        random_state=self.random_state, solver='saga')
```

**Comment by marcotcr** [comment]
Does it matter? Have you seen examples where this doesn't behave sensibly?
Thanks,

**Comment by 0shimax** [comment]
Yes, it matters.
If deep learning is used as a model and gradient explosion occurs (in case can not learn without gradient clip), SVD cannot be solved and errors tend to occur.

**Comment by marcotcr** [comment]
How about for ridge, as is our case?

**Comment by 0shimax** [comment]
I mean that I use DNN as a predict_func in "explain_instance" function, just in case.
For Ridge, you can see the description below in documentation of Scikit-learn.

> All last five solvers support both dense and sparse data. However, only ‘sag’ and ‘saga’ supports sparse input when fit_intercept is True.

http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.Ridge.html

**Comment by marcotcr** [comment]
It seems to me that the `predict_fn` does not matter at all at this point, as it just provides the labels that Ridge will use. SVD is being used to optimize ridge, not whatever `predict_fn` is. Am I missing something?
